test/e2e/e2e_test.go

```diff
@@ -81,3 +85,40 @@ var _ = Describe("Workload cluster creation", func() {
 		})
 	})
 })
+
+func updateCalico(calicoYaml, calicoInterface string) {
+	err := downloadFile(calicoYaml, "https://docs.projectcalico.org/manifests/calico.yaml")
+	Expect(err).To(BeNil(), "Unable to download Calico manifest")
+	cniYaml, err := os.ReadFile(calicoYaml)
+	Expect(err).To(BeNil(), "Unable to read Calico manifest")
+	podCIDR := os.Getenv("POD_CIDR")
+	cniYaml = []byte(strings.Replace(string(cniYaml), "192.168.0.0/16", podCIDR, -1))
```

Comment:
I am not getting, what are we after here?

Reply:
I am trying to archive the same thing as: https://github.com/metal3-io/metal3-dev-env/blob/137e648c117cf7058f921e4fa550ef94cce620d3/vm-setup/roles/v1aX_integration_test/tasks/verify.yml#L68
However, I start to doubt if this change has any impact because the replacement takes place in a comment. https://github.com/metal3-io/cluster-api-provider-metal3/blob/6abef24bfd6a97643a0a3cf7ed41fdbb4352f431/test/e2e/data/cni/calico/calico.yaml#L3684
Should we remove this change?

Reply:
I am fine leaving it as is to be consistent with m3-dev-env task.